lib/src/realm_class.dart

```diff
@@ -935,3 +892,154 @@ class MigrationRealm extends DynamicRealm {
 /// * syncProgress - an object of [SyncProgress] that contains `transferredBytes` and `transferableBytes`.
 /// {@category Realm}
 typedef ProgressCallback = void Function(SyncProgress syncProgress);
+
+/// Specifies the criticality level above which messages will be logged
+/// by the default sync client logger.
+///
+/// {@category Realm}
+class RealmLogLevel {
+  /// Log everything. This will seriously harm the performance of the
+  /// sync client and should never be used in production scenarios.
+  ///
+  /// Same as [Level.ALL]
+  static const all = Level.ALL;
+
+  /// A version of [debug] that allows for very high volume output.
+  /// This may seriously affect the performance of the sync client.
+  ///
+  /// Same as [Level.FINEST]
+  static const trace = Level('TRACE', 300);
+
+  /// Reveal information that can aid debugging, no longer paying
+  /// attention to efficiency.
+  ///
+  /// Same as [Level.FINER]
+  static const debug = Level('DEBUG', 400);
+
+  /// Same as [info], but prioritize completeness over minimalism.
+  ///
+  /// Same as [Level.FINE];
+  static const detail = Level('DETAIL', 500);
+
+  /// Log operational sync client messages, but in a minimalist fashion to
+  /// avoid general overhead from logging and to keep volume down.
+  ///
+  /// Same as [Level.INFO];
+  static const info = Level.INFO;
+
+  /// Log errors and warnings.
+  ///
+  /// Same as [Level.WARNING];
+  static const warn = Level.WARNING;
+
+  /// Log errors only.
+  ///
+  /// Same as [Level.SEVERE];
+  static const error = Level('ERROR', 1000);
+
+  /// Log only fatal errors.
+  ///
+  /// Same as [Level.SHOUT];
+  static const fatal = Level('FATAL', 1200);
+
+  /// Turn off logging.
+  ///
+  /// Same as [Level.OFF];
+  static const off = Level.OFF;
+}
+
+/// Represents a logger that manages the realm log level at runtime.
+///
+/// {@category Realm}
+class RealmLogger implements Logger {
+  final Logger _logger;
+
+  /// Creates an instance of [RealmLogger].
+  RealmLogger({Level level = RealmLogLevel.info, void Function(LogRecord)? onRecord})
```

Comment:
I think we should get rid of this ctor, and set the default logger on the static `Realm.logger` property instead, and have it wrapped by a `_RealmLogger` just as if the user had set it themself.